parser/parser.go

```diff
@@ -0,0 +1,52 @@
+package parser
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/reviewdog/errorformat/fmts"
+	"github.com/reviewdog/reviewdog/proto/rdf"
+)
+
+// Parser is an interface which parses compilers, linters, or any tools
+// results.
+type Parser interface {
+	Parse(r io.Reader) ([]*rdf.Diagnostic, error)
+}
+
+// ParserOpt represents option to create Parser. Either FormatName or
+// Errorformat should be specified.
+type ParserOpt struct {
```

Comment:
**[golint-circleci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
type name will be used as parser.ParserOpt by other packages, and that stutters; consider calling this Opt